sdk/batch/batch/src/batchSharedKeyCredentials.ts

```diff
@@ -64,17 +64,21 @@ export class BatchSharedKeyCredentials implements ServiceClientCredentials {
     }
 
     // Help function to get content length
-    function getContentLengthToAppend(value: HttpHeaders, method: HttpMethods, body: any): string {
+    function getContentLengthToAppend(
+      value: HttpHeadersLike,
+      method: HttpMethods,
+      body: any
+    ): string {
       if (!value || !value.get("Content-Length")) {
         // Get content length from body if available
         if (body) {
           return Buffer.byteLength(body) + "\n";
         }
         // For GET verb, do not add content-length
-        if (method === "GET") {
-          return "\n";
-        } else {
+        if (method === "POST") {
           return "0\n";
+        } else {
+          return "\n";
```

Comment:
A description of how the authorization header is created can be found [here](https://docs.microsoft.com/en-us/rest/api/batchservice/authenticate-requests-to-the-azure-batch-service).